### ISSUE TITLE
broken soundcloud auth flow - window doesn't close after authenticated

### ISSUE THREAD

**Issue by dryajov** [soundcloud/soundcloud-javascript#76]
After authenticating with soundcloud with the popup flow, the popup never closes and just sits there with `This popup should automatically close in a few seconds` message. I see the following stacktrace in the console:

```
Uncaught TypeError: Cannot read property 'search' of undefined
    at Object.notifyDialog (/scripts/main.js:97393)
    at Object.connectCallback (/scripts/main.js:97386)
    at soundcloud_auth.html:9
notifyDialog @ /scripts/main.js:97393
connectCallback @ /scripts/main.js:97386
(anonymous) @ soundcloud_auth.html:9
```

PR: #47 fixes this particular issue. I can consistently reproduce this issue on:

- Safari: Version 10.0.1 (12602.2.14.0.7)
- Chrome: Version 55.0.2883.95 (64-bit)
- Firefox: Version 50.1.0

**Comment by SergeyBondarenko** [comment]
@dryajov I have broken authentication flow too. Please look the issue #77, do you think it is related to your issue?

**Comment by dryajov** [comment]
@SergeyBondarenko that sounds more like a CORS issue, can you make sure that the redirect url (might be called differently in SC) in your soundcloud account is set to the exact host you're authenticating from? Protocol, host name and port should all match.

**Comment by SergeyBondarenko** [comment]
@dryajov yes, I checked it already, the redirect URI in my code is the same as the registered one and I'm trying to sign-in from the same `protocol://domain.com`

**Comment by dryajov** [comment]
@SergeyBondarenko That is weird, but still seem to be a different issues, just from the call stack you provided. 

Is that over http or https? There might also be a restriction with iframes cross-domain insecure http calls in some browsers, I know that chrome has been cracking down on http lately, might be worth trying it in other browsers and/or looking at iframe restrictions in general.

**Comment by ldiqual** [comment]
Fyi, this is how I fixed it:

```
window.opener.SC.location = window.location
window.opener.SC.connectCallback()
delete window.opener.SC.location
```

However https://github.com/soundcloud/soundcloud-javascript/pull/47 seems to have a proper fix.
